sway-lsp/src/capabilities/code_actions/abi_decl/mod.rs

```diff
@@ -0,0 +1,21 @@
+pub(crate) mod abi_impl;
+
+use sway_core::decl_engine::DeclId;
+use sway_types::Spanned;
+use tower_lsp::lsp_types::CodeActionOrCommand;
+
+use self::abi_impl::AbiImplCodeAction;
+
+use super::{CodeActionContext, CodeActionTrait};
+
+pub(crate) fn code_actions(
+    decl_id: &DeclId,
+    ctx: CodeActionContext,
+) -> Option<Vec<CodeActionOrCommand>> {
+    let decl = ctx
+        .engines
+        .de()
+        .get_abi(decl_id.clone(), &decl_id.span())
+        .unwrap();
```

Comment:
unwrap...

Reply:
I tried changing this to `let decl = ctx.engines.de().get_abi(decl_id.clone(), &decl_id.span())?;`  but I get an error: 

```
the `?` operator can only be used on `Option`s, not `Result`s, in a function that returns `Option`
the trait `FromResidual<Result<std::convert::Infallible, CompileError>>` is not implemented for `std::option::Option<Vec<CodeActionOrCommand>>`
the following other types implement trait `FromResidual<R>`:
  <std::option::Option<T> as FromResidual<Yeet<()>>>
  <std::option::Option<T> as FromResidual>rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fsophiedankel%2FDevelopment%2Fsway%2Fsway-lsp%2Fsrc%2Fcapabilities%2Fcode_actions%2Fabi_decl%2Fmod.rs)
mod.rs(11, 1): this function returns an `Option`
```

Is there a better way?

Reply:
I'm not sure why it works with `?` in other places but not here.

Reply:
`.ok()?` worked.